department-of-veterans-affairs/caseflow
# CASEFLOW-3717: VSO Convert to Virtual Form Validation

Resolves [CASEFLOW-3717](https://vajira.max.gov/browse/CASEFLOW-3717)

### Description
Real time email/confirm validation and checkbox affirmation disables "Convert to Virtual" button. An action to "Convert to Virtual" will not POST if: Timezone field is null and Email Confirmation field is an empty string. Appropriate error messages are shown to the user based on the states of the incorrect/missing fields. 

### Acceptance Criteria 
- [ ] The selectable "Convert to Virtual Hearing" option is disabled unless all required fields are populated Required fields:
- 	Veteran/Appellant Email (for hearings notifications only)
- 	Confirm Veteran / Appellant Email
- 	Veteran / Appellant Timezone
- 	POA/Representative Timezone
- 	Following Affirmations are NOT Toggled (Affirmation Toggle Validation) 
- 	I affirm that I have the Veteran / Appellant's permission to opt-in to a virtual hearing
- I affirm that the Veteran / Appellant and POA / Representative will have access to a computer, table, or mobile device with a realizable interned connection, camera

- [ ] As user enters an email into the Appellant/Veteran email field they are given a warning if the email isn't valid.
- [ ] User is informed when the following email inputs do not match after entering at least one character into the confirm email field and unselecting the field.

### Testing Plan 
1) Intake a new appeal under a hearing 
2) Take note of uuid after successfully moving through intake 
3) Switch to user: BILLIE_VSO 
4) Navigate to url: /queue/appeals/[uuid]

The below steps are dependent on https://github.com/department-of-veterans-affairs/caseflow/pull/17269

5) Click on action dropdown and select "Convert to virtual"
6) Confirm that "Convert to Virtual" button is disabled 
7) Confirm email is valid. If not, error message is thrown. 
8) Confirm that 'confirm email' field matches original email. If not, error message is thrown. 
9) Enter Timezones for both the appellant and representative. 
10) Affirm checkboxes 
11) Confirm that "Convert to Virtual" button is now enabled.
12) Confirm renavigation to "Case Details" page.
